api/v1beta1/overcloudnet_types.go

```diff
@@ -0,0 +1,75 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// IPReservation contains an IP, and Hostname
+type IPReservation struct {
+	//IP       net.IP `json:"ip"`
+	IP       string `json:"ip"`
+	Hostname string `json:"hostname,omitempty"`
+}
+
+// OvercloudNetSpec defines the desired state of OvercloudNet
+type OvercloudNetSpec struct {
+
+	// +kubebuilder:validation:Required
+	// Cidr the cidr to use for this network
+	Cidr string `json:"cidr"`
+
+	// +kubebuilder:validation:Required
+	// AllocationStart a set of IPs that are reserved and will not be assigned
+	AllocationStart string `json:"allocationStart,omitempty"`
```

Comment:
Just curious here...are the `// +kubebuilder:validation:Required` and `omitempty` competing with one another?  My understanding was that `omitempty` made a field optional.

Reply:
yeah, let me revisit those validations. When I initially created the APIs I was thinking those would be optional but after implementing it I think we probably do just want to require them.

Reply:
@abays fixed in the latest push